### ISSUE TITLE
训练竖排检测模型时，MTHv2 全量数据 15 epoch 后 hmean 仅 0.045，请教下一步如何调整

### ISSUE THREAD

**Issue by funnyzak** [PaddlePaddle/PaddleOCR#17901]
### 🔎 Search before asking

- [x] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [x] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [x] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### 🐛 Bug (问题描述)

当前目标是基于 `MTHv2/ TKHMTH2200` 数据做古籍竖排 OCR 微调，最终用于古籍竖排文本识别。

目前数据集已经下载并整理好了，训练链路是先做文本检测，再继续做文本识别。当前第一步就卡住了：**古籍整页中的竖排正文列检测训练效果很差。**

当前使用的是 PaddleOCR 的训练脚本和配置：

- 训练脚本：`tools/train.py`
- 检测配置：`configs/det/PP-OCRv5/PP-OCRv5_mobile_det.yml`
- 预训练权重：`https://paddleocr.bj.bcebos.com/pretrained/PPLCNetV3_x0_75_ocr_det.pdparams`

当前使用的数据集，是将 `MTHv2 / TKHMTH2200` 原始数据转换成 PaddleOCR 检测训练格式后得到的全量数据。训练过程本身没有报错，loss 也在持续下降，但验证集指标很差，到第 `15` 个 epoch 时 best `hmean` 只有 `0.0451`，其中 `recall=0.0247`，完全达不到可用状态。

### 环境信息

- 操作系统：`Windows 11`
- GPU：`NVIDIA GeForce RTX 4070`
- 系统内存：`192 GB`
- CUDA：`12.9`
- cuDNN：`9`
- Paddle 版本： `paddle 3.2.0`
- 训练方式：单卡 GPU

`nvcc -V` 输出：

```text
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2025 NVIDIA Corporation
Built on Wed_Apr__9_19:29:17_Pacific_Daylight_Time_2025
Cuda compilation tools, release 12.9, V12.9.41
Build cuda_12.9.r12.9/compiler.35813241_0
```

当前机器上查到的 CUDA / cuDNN 动态库如下：

```text
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\cudnn64_9.dll
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\cudnn_adv64_9.dll
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\cudnn_cnn64_9.dll
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\cudnn_engines_precompiled64_9.dll
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\cudnn_engines_runtime_compiled64_9.dll
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\cudnn_graph64_9.dll
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\cudnn_heuristic64_9.dll
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\cudnn_ops64_9.dll
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\cublas64_12.dll
```

### 数据集说明

原始 `MTHv2 / TKHMTH2200` 已转换为 PaddleOCR 检测训练格式，目录如下：

```text
mthv2-paddleocr-dataset/
├── guji_raw/
├── guji_det/
│   ├── images/
│   │   ├── train/
│   │   ├── val/
│   │   └── test/
│   ├── train_label.txt
│   ├── val_label.txt
│   └── test_label.txt
├── guji_rec/
└── meta/
```

检测任务相关统计如下：

- 总页数：`3199`
- 训练页数：`2559`
- 验证页数：`320`
- 测试页数：`320`
- 检测文本行总数：`105579`
- 训练检测文本行数：`76028`
- 验证检测文本行数：`13218`
- 测试检测文本行数：`16333`
- `crop_failures=0`
- `malformed_textlines=0`
- 空文本检测标签：`1` 条，保留为 `###`

当前检测标注粒度以整列或整行为主，伴随少量页眉、题头、短文本框或单字框。绝大多数框仍然是高而窄的竖排长文本列。按训练集统计：

- `train` 中框高宽比 `h/w` 中位数约为 `10.25`
- `train` 中约 `91.0%` 的框满足 `h/w > 1.5`
- `train` 中框宽度中位数约 `93 px`
- `train` 中框高度中位数约 `1011 px`

整页图分辨率也比较大：

- `train` 页图宽度中位数约 `2400 px`，高度中位数约 `2295 px`
- `val` 页图宽度中位数约 `2417 px`，高度中位数约 `3424 px`
- `test` 页图宽度中位数约 `2466 px`，高度中位数约 `3456 px`

### 训练方式

当前直接使用 PaddleOCR 的 det 训练方式，只改了数据路径、训练轮数、学习率、batch 和 workers。

配置文件：

- `configs/det/PP-OCRv5/PP-OCRv5_mobile_det.yml`

主要覆盖参数：

- `Global.epoch_num=15`
- `Optimizer.lr.learning_rate=0.00005`
- `Train.loader.batch_size_per_card=4`
- `Train.loader.num_workers=0`
- `Eval.loader.num_workers=0`

训练命令如下：

```powershell
python tools/train.py -c configs/det/PP-OCRv5/PP-OCRv5_mobile_det.yml `
  -o Global.save_model_dir=./output/guji_det_mobile_full `
     Global.epoch_num=15 `
     Optimizer.lr.learning_rate=0.00005 `
     Train.dataset.data_dir="$DataRootFull/guji_det" `
     Train.dataset.label_file_list="['$DataRootFull/guji_det/train_label.txt']" `
     Eval.dataset.data_dir="$DataRootFull/guji_det" `
     Eval.dataset.label_file_list="['$DataRootFull/guji_det/val_label.txt']" `
     Train.loader.batch_size_per_card=4 `
     Train.loader.num_workers=0 `
     Eval.loader.num_workers=0
```

训练日志中的关键信息：

- `train with paddle 3.2.0 and device Place(gpu:0)`
- `train dataloader has 640 iters`
- `Global.pretrained_model` 为 `https://paddleocr.bj.bcebos.com/pretrained/PPLCNetV3_x0_75_ocr_det.pdparams`

日志里还有以下预训练权重 warning：

```text
The pretrained params backbone.last_conv.weight not in model
The pretrained params backbone.act.lab.scale not in model
The pretrained params backbone.act.lab.bias not in model
The pretrained params backbone.fc.weight not in model
The pretrained params backbone.fc.bias not in model
```

### 训练结果

训练过程中 loss 在持续下降，例如：

- `global_step=100` 时，`loss=8.1204`
- `global_step=1500` 时，`loss=4.1879`
- `global_step=3000` 时，`loss=3.3057`
- `global_step=6000` 时，`loss=2.5130`
- `global_step=9000` 时，`loss=2.2467`

但验证集指标始终很差，关键评估点如下：

| epoch | precision | recall | hmean |
| --- | ---: | ---: | ---: |
| 3 | 0.0000 | 0.0000 | 0.0000 |
| 5 | 0.0000 | 0.0000 | 0.0000 |
| 8 | 0.0108 | 0.000228 | 0.000446 |
| 10 | 0.0762 | 0.00410 | 0.00779 |
| 12 | 0.2180 | 0.01489 | 0.02788 |
| 15 | 0.2586 | 0.02469 | 0.04508 |

最终 best metric：

```text
best metric, hmean: 0.04507628294036061
precision: 0.25855210819411295
recall: 0.024690420116994606
best_epoch: 15
```

### 已确认

- 数据已经转成 PaddleOCR det 的常见标注格式：`图片相对路径<TAB>JSON数组`
- 训练输入是整页图，不是切碎后的单字图
- 标注粒度以整列或整行文本区域为主，不是切碎成单字检测数据集
- 训练过程没有出现 `OOM`、`NaN`、数据解析报错或中断

### 希望进一步确认的问题

1. 对于这种古籍竖排长文本列场景，当前检测标注粒度是否正确？

   当前采用的是以整列或整行为主的文本区域框，同时包含少量短框或单字框。

2. 对于这种大分辨率整页图 + 超高竖列框场景，默认 `PP-OCRv5_mobile_det` 配置是否不合适？

3. 对这种全量数据来说，`15 epoch` 是否明显偏少？

   如果主要问题是训练轮数不够，希望得到一个更合理的轮数范围，或者优先应该调整的参数方向。

4. 上面那几条预训练权重 warning 是否可以完全忽略？

5. 或者说下一步应该从哪个方向调整，以提供训练效果？

### 相关附件

建了个临时 Repo [ppaddleocr-mthv2-vertical-det-issue](https://github.com/funnyzak/paddleocr-issues/tree/main/mthv2-vertical-det-issue)，方便进一步排查。内容包括与此相关的：

- 完整训练日志
- 环境与训练命令
- 数据集目录结构
- 数据统计摘要
- 选定页面样例图
- 对应页面的完整标签
- 标注可视化图

### 🏃‍♂️ Environment (运行环境)

- 操作系统：`Windows 11`
- GPU：`NVIDIA GeForce RTX 4070`
- 系统内存：`192 GB`
- CUDA：`12.9`
- cuDNN：`9`
- Paddle 版本： `paddle 3.2.0`
- 训练方式：单卡 GPU

### 🌰 Minimal Reproducible Example (最小可复现问题的Demo)

```powershell
python tools/train.py -c configs/det/PP-OCRv5/PP-OCRv5_mobile_det.yml `
  -o Global.save_model_dir=./output/guji_det_mobile_full `
     Global.epoch_num=15 `
     Optimizer.lr.learning_rate=0.00005 `
     Train.dataset.data_dir="$DataRootFull/guji_det" `
     Train.dataset.label_file_list="['$DataRootFull/guji_det/train_label.txt']" `
     Eval.dataset.data_dir="$DataRootFull/guji_det" `
     Eval.dataset.label_file_list="['$DataRootFull/guji_det/val_label.txt']" `
     Train.loader.batch_size_per_card=4 `
     Train.loader.num_workers=0 `
     Eval.loader.num_workers=0
```

**Comment by alrowilde** [comment]
To be honest, It is not an ideal dataset for vertical detection task.

The labeling approach is not suitable for paddleOCR, because of some texts are unlabeledm and haven't splitting the line into two parts when the texts have a great of blank spacing.

You can see this example to confirm the problems there.
https://github.com/funnyzak/paddleocr-issues/blob/main/mthv2-vertical-det-issue/samples/visualized/test__MTH1000__85-V095P0199__vis.png

**Comment by funnyzak** [comment]
感谢指出问题所在。
还是对 PaddleOCR 的认识不足 😊。
